### PR TITLE
Add env var to disable creation of PodDisruptionBudget

### PR DESCRIPTION
**Pull request by katheris**
### Type of change

_Select the type of your PR_

- Enhancement / new feature

### Description

Add env var to disable creation of PodDisruptionBudget

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Write tests
- [x] Make sure all tests pass
- [ ] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [x] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [x] Update CHANGELOG.md
- [ ] Supply screenshots for visual changes, such as Grafana dashboards

### CODE REVIEW

File: cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/ZooKeeperReconciler.java [hunk 1]
```diff
@@ -87,6 +87,7 @@ public class ZooKeeperReconciler {
     private final int adminSessionTimeoutMs;
     private final ImagePullPolicy imagePullPolicy;
     private final List<LocalObjectReference> imagePullSecrets;
+    private final boolean isPodDisruptionBudgetGeneration;
```
Comment (scholzj) [inline cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/ZooKeeperReconciler.java:90]: Maybe we can keep it together with the `isNetworkPolicyGeneration`?

File: cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaReconciler.java [hunk 2]
```diff
@@ -122,6 +122,7 @@ public class KafkaReconciler {
     private final ImagePullPolicy imagePullPolicy;
     private final List<LocalObjectReference> imagePullSecrets;
     private final List<Integer> previousNodeIds;
+    private final boolean isPodDisruptionBudgetGeneration;
```
Comment (scholzj) [inline cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaReconciler.java:125]: Maybe we can keep it together with the `isNetworkPolicyGeneration`?

File: cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/AbstractAssemblyOperator.java [hunk 3]
```diff
@@ -63,6 +63,7 @@ public abstract class AbstractAssemblyOperator<C extends KubernetesClient, T ext
     protected final List<LocalObjectReference> imagePullSecrets;
     protected final KafkaVersion.Lookup versions;
     protected long operationTimeoutMs;
+    protected final boolean isPodDisruptionBudgetGeneration;
```
Comment (scholzj) [inline cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/AbstractAssemblyOperator.java:66]: Is it worth having it here? `KafkaMirrorMakerAssembly` will be deleted soon. So there will be just two classes inheriting from this - one using this field and one not using it. Maybe it is better to set this directly in KafkaBridgeAssemblyOperator and KAfkaMirrorMakerAssemblyOperator (unless we decide to ignore it completely given it has only few months to live)?

**Review by scholzj — COMMENTED** [review]
Some nits, but LGTM otherwise.

**Review by scholzj — COMMENTED** [review]
Oh, one more thing ... can you add it to the docs as well ... probably here where the network policy env var is already documented? https://strimzi.io/docs/operators/latest/full/deploying.html#ref-operator-cluster-str

File: cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/AbstractAssemblyOperator.java (hunk 3 above)
Comment (katheris) [reply cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/AbstractAssemblyOperator.java]: Good point, I've removed it from `KafkaMirrorMakerAssembly` completely

**Comment by katheris** [comment]
> Oh, one more thing ... can you add it to the docs as well ... probably here where the network policy env var is already documented? https://strimzi.io/docs/operators/latest/full/deploying.html#ref-operator-cluster-str

I chatted to Paul and added it in two places, there and in a section in the configuring part of the docs.

**Review by scholzj — APPROVED** [review]
LGTM. Thanks.

File: documentation/modules/configuring/proc-disable-pod-disruption-budget-generation.adoc [hunk 4]
```diff
@@ -0,0 +1,11 @@
+// Module included in the following assemblies:
+//
+// assembly-config.adoc
+
+[id='disable-pod-disruption-budget-generation_{context}']
+= Disabling pod disruption budget generation
+
+Strimzi generates pod disruption budget resources for Kafka, Kafka Connect worker, MirrorMaker2 worker, and Kafka Bridge worker nodes.
+
+If you want to use custom pod disruption budget resources, you can set the `STRIMZI_POD_DISRUPTION_BUDGET_GENERATION` environment variable to `false` in the Cluster Operator configuration.
```
Comment (PaulRMellor) [inline documentation/modules/configuring/proc-disable-pod-disruption-budget-generation.adoc:10]:
```suggestion
If you don't want to use custom pod disruption budget resources, you can set the `STRIMZI_POD_DISRUPTION_BUDGET_GENERATION` environment variable to `false` in the Cluster Operator configuration.
```

**Review by PaulRMellor — APPROVED** [review]
Great. Just one comment.

File: documentation/modules/configuring/proc-disable-pod-disruption-budget-generation.adoc (hunk 4 above)
Comment (katheris) [reply documentation/modules/configuring/proc-disable-pod-disruption-budget-generation.adoc:10]: By "custom pod disruption budget resources" I meant if the user wants to provide their own, so for that they need to disable the generation, so I think the existing text is correct

**Review by ppatierno — APPROVED** [review]
LGTM

**Comment by scholzj** [comment]
@katheris Could you please rebase it?

**Comment by katheris** [comment]
> @katheris Could you please rebase it?

Done :)

**Comment by scholzj** [comment]
Thanks for the PR @katheris

**Pull request merged**